src/document/DocumentAnnotator.ts

```diff
@@ -78,4 +84,55 @@ export default class DocumentAnnotator extends BaseAnnotator {
 
         this.managers.set(pageNumber, pageManagers);
     }
+
+    scrollTo(page = 1, centerX = 0, centerY = 0): void {
+        const pageEl = this.getPage(page);
+
+        if (!this.annotatedEl || !pageEl) {
+            return;
+        }
+
+        const canSmoothScroll = 'scrollBehavior' in this.annotatedEl.style;
+        const parentCenterX = this.annotatedEl.clientWidth / 2;
+        const parentCenterY = this.annotatedEl.clientHeight / 2;
+        const shapeCenterX = Math.round(centerX * this.scale); // prettier-ignore
+        const shapeCenterY = Math.round(centerY * this.scale); // prettier-ignore
+        const shapeScrollLeft = pageEl.offsetLeft - parentCenterX + shapeCenterX;
+        const shapeScrollTop = pageEl.offsetTop - parentCenterY + shapeCenterY;
+        const scrollLeft = Math.max(0, Math.min(shapeScrollLeft, this.annotatedEl.scrollWidth));
+        const scrollTop = Math.max(0, Math.min(shapeScrollTop, this.annotatedEl.scrollHeight));
+
+        if (canSmoothScroll && Math.abs(this.annotatedEl.scrollTop - scrollTop) < 1000) {
+            this.annotatedEl.scrollTo({
+                behavior: 'smooth',
+                left: scrollLeft,
+                top: scrollTop,
+            });
+        } else {
+            this.annotatedEl.scrollLeft = scrollLeft;
+            this.annotatedEl.scrollTop = scrollTop;
+        }
+    }
+
+    setActiveId(annotationId: string | null): void {
+        if (annotationId) {
+            const annotation = getAnnotation(this.store.getState(), annotationId);
+            const page = annotation?.target.location.value || 1;
+
+            if (!annotation) {
+                return;
+            }
+
+            if (annotation.target.type === 'region') {
+                const { height, width, x, y } = annotation.target.shape;
+                const centerX = (x + (width / 2)); // prettier-ignore
+                const centerY = (y + (height / 2)); // prettier-ignore
+
+                this.scrollTo(page, centerX, centerY);
+            }
+        }
```

Comment:
Cool, this should also fix an issue we have right now where clicking on an annotation that is already an active annotation would not scroll that annotation into view